### ISSUE TITLE
[Bug] Keyboard shortcut to toggle DR v4.9.37.1 not working

### ISSUE THREAD

**Issue by Gitoffthelawn** [darkreader/darkreader#6891]
# Bug Report

## Bug Description
Keyboard shortcut to toggle DR v4.9.37.1 not working.

## Website Address
All

## Steps To Reproduce
1. Upgraded DR to v4.9.37.1
2. Restarted Firefox
3. Pressed keyboard shortcut to toggle DR - it toggled correctly
4. Pressed keyboard shortcut to toggle DR again - nothing happened
5. Repeated step 4 - still nothing
6. Restarted Firefox
7. Repeated step 4 - still nothing

## Expected Behavior
Keyboard shortcut to toggle DR to, um, toggle DR

## Actual Behavior
See above

## System Information

- OS: Windows
- Browser: Firefox 92.0.1
- Dark Reader Version: 4.9.37.1

## Additional Context
I do have a custom keyboard shortcut defined for this behaviour in `about:addons`.

**Comment by bershanskiy** [comment]
I am unable to reproduce this. Shortcuts work fine for me. Will continue testing.

**Comment by Altgear** [comment]
Can confirm toggling DR on/off stops working on Ubuntu.
Toggling for current site seems to work, however DR resets this setting after page refresh.
Downgrading to DR 4.9.34 fixed everything for me.

> **System information**

- OS: xubuntu 18.04.05 LTS 
- Browser: Firefox 90.0
- Dark Reader Version: 4.9.37.1

**Comment by Gusted** [comment]
Okay I'm finally able to reproduce this, I clearly can see in the UI that the changes are coming trough but that no message is sent to the tab? to actually do something with these changes.

**Comment by Gusted** [comment]
Error: ` TypeError: can't access property "timestamp", this.tabs[tab.id][frameId] is undefined`:
https://github.com/darkreader/darkreader/blob/910ff505d93c4078a4a08710d0b1bcc9d1608287/src/background/tab-manager.ts#L275

@bershanskiy This is a regression from your refactor PR. Given that this is not in v4.9.37.1 Il have to test it again to see which error this Bug is causing.

**Comment by Gusted** [comment]
Hmm pretty weird, on v4.9.37.1 it's just working correctly. 🤷🏽

**Comment by Gitoffthelawn** [comment]
> Hmm pretty weird, on v4.9.37.1 it's just working correctly. 🤷🏽

Are you saying it's now working for you on v4.9.37.1?  For me, it's not.

**Comment by Gusted** [comment]
> Are you saying it's now working for you on v4.9.37.1? For me, it's not.

Yes it's working for me. It was a regression caused by a little refactor which was now reverted and it should be the old behavior.

**Comment by Gitoffthelawn** [comment]
> > Are you saying it's now working for you on v4.9.37.1? For me, it's not.
> 
> Yes it's working for me. It was a regression caused by a little refactor which was now reverted and it should be the old behavior.

That was my understand too.  But for v4.9.37.1, it's surprisingly still broken for me.  Maybe I'll try reinstalling the extension.  Technically, it shouldn't matter, but it's fairly quick to try.

I'll check logs before doing so and report back if I see anything odd.

**Comment by Gitoffthelawn** [comment]
I checked logs... nothing.  But toggling DR via the popup doesn't work reliably either, so I'm not sure if this is really an effect of https://github.com/darkreader/darkreader/issues/6873 or a separate issue.

Toggling DR via the popup works sometimes, but not others... it's real odd behaviour.

I might wait until next release instead of backing up / uninstalling / reinstalling / restoring... we'll see what time permits...

**Comment by Gitoffthelawn** [comment]
Well, I backed up DR, uninstalled, reinstalled, restored.

After doing this, unfortunately nothing improved.  DR keyboard shortcut not working reliably and even the toggle in the DR popup not working reliably.  :(

**Comment by Gusted** [comment]
> After doing this, unfortunately nothing improved. DR keyboard shortcut not working reliably and even the toggle in the DR popup not working reliably. :(

Hmm then we might have another regression which I find weird. Could you try this on a new profile maybe? Otherwise you will have to run a git-bisect to see if you can locate the commit which introduced this regression as I'm still not able to reproduce this behavior.

**Comment by Gitoffthelawn** [comment]
> Hmm then we might have another regression which I find weird. Could you try this on a new profile maybe? Otherwise you will have to run a git-bisect to see if you can locate the commit which introduced this regression as I'm still not able to reproduce this behavior.

Yes, I will work on this some more. Very busy right now, but I'll do my best to squeeze in some time to work on this.

I have a hunch it's all related to #6873.  Once that has a working patch, I'll be able to quickly figure out if this is a side-effect.

**Comment by Altgear** [comment]
I did a little testing of different versions on new profile and here are the results:

- Version 4.9.34 and before: everything works as intended
- Version 4.9.35: all toggle shortcuts don't work ,toggling DR on/off via popup works only once, toggling via popup for current site works as many times as you want, but only until you toggle DR off once. As a result you get a totally unresponsive DR. 
- Version 4.9.36: all toggle shortcuts do work at first, though, site settings aren't preserved after page refresh or toggling DR on/off.
 !!! **BUT** after toggling DR on/off via popup once all toggle shortcuts stop working again, just like in version 4.9.35 !!!
 I also get same broken behavior as I mentioned above after trying to toggle DR for specific site via shortcut when DR is turned off...
- Version 4.9.37.1: same as 4.9.36

My point is: This is not a issue in just 4.9.37.1, but also 4.9.36 or even before that, since the last working as intended version is 4.9.34.
Hope it helps you a little...
> > **System information**
> 
>     * OS: xubuntu 18.04.05 LTS
> 
>     * Browser: Firefox 90.0
> 
>     * Dark Reader Version: 4.9.37.1

**Comment by Gusted** [comment]
Hmm, it seems to be related to #6873 then. Just leaving it open for now until that issue is fixed and the release with that fix is released.

**Comment by Gitoffthelawn** [comment]
@Altgear Thank you for all that testing!

Everything you wrote exactly correlates with my recollection.
